### PR TITLE
output: Implement adaptive sync option

### LINKED ISSUE #1009: Add support for variable refresh rate / adaptive sync
Since wlroots added support for this 6 months ago, now may be a good time to add an output option to actually enable or disable it.

See: wlr_output_enable_adaptive_sync

### PR DESCRIPTION
**Pull request by soreau**
Enable by setting [output:<NAME>] vrr = true, or using a client like wlr-randr.

Fixes #1009.

### CODE REVIEW

File: src/core/output-layout.cpp [hunk 1]
```diff
@@ -581,6 +586,29 @@ struct output_layout_output_t
...
+            {
+                wlr_output_enable_adaptive_sync(handle, current_state.vrr);
+                wlr_output_commit(handle);
+                if (handle->adaptive_sync_status == WLR_OUTPUT_ADAPTIVE_SYNC_DISABLED)
+                {
+                    LOGE("Failed to enable adaptive sync on output: ", handle->name);
+                    wlr_output_enable_adaptive_sync(handle, false);
+                } else
+                {
+                    LOGI("Enabled adaptive sync on output: ", handle->name);
```
Comment (ammen99) [inline src/core/output-layout.cpp:601]: Sounds like a good candidate for LOGD, not LOGI. Optionally, maybe even add a new logging category to debugging.hpp (output-related)?

File: src/core/output-layout.cpp [hunk 2]
```diff
@@ -581,6 +586,29 @@ struct output_layout_output_t
             wlr_output_set_custom_mode(handle, mode.width, mode.height, mode.refresh);
         }
 
+        if (!is_nested_compositor)
```
Comment (ammen99) [inline src/core/output-layout.cpp:589]: Is this something wlroots doesn't support or why put this check?

**Review by ammen99 — COMMENTED** [review]
As a whole LGTM, two small nitpicks.

File: src/core/output-layout.cpp (hunk 2 above)
Comment (soreau) [reply src/core/output-layout.cpp:589]: When I use `if  (1)` here, nested wayland backend does not start. it fails with:
```
[types/output/swapchain.c:109] Swapchain for output 'WL-1' failed test
```

File: src/core/output-layout.cpp (hunk 1 above)
Comment (soreau) [reply src/core/output-layout.cpp:601]: Sure, we can make it LOGD but I'm not sure it warrants a new debug category.

By the way, is there a way to enable all debug categories at once with a single option?

File: src/core/output-layout.cpp (hunk 2 above)
Comment (soreau) [reply src/core/output-layout.cpp]: I think I fixed it without the special case.

**Review by ammen99 — APPROVED** [review]
Just pushed a small commit to reduce duplicated logic, otherwise LGTM. Thanks!

**Pull request merged**